### ISSUE TITLE
Structure files 

### ISSUE THREAD

**Issue by vsumaria** [MDIL-SNU/SIMPLE-NN_v2#73]
Is there a possibility to use extended XYZ or ase trajectory files to read the structural energy and force data instead of OUTCAR? 

In the data_processing.py, you load_structures function is creating a ase atoms object, so I think there should be a way to directly read files other than OUTCAR with ase for the E, F, Position data right?

**Comment by JisuJung928** [comment]
Hello, vsumaria.

I fixed the format bug immediately so that up-to-date SIMPLE-NN can handle the various format including extended XYZ. 
I am sorry for the inconvenience.

Please update your SIMPLE-NN and enjoy the training!

Best regards,
Jisu.

**Comment by vsumaria** [comment]
Thanks @JisuJung928. 

To run the code in parallel (using mpi), do I need to include any specific changes in the input.yaml ?

**Comment by JisuJung928** [comment]
You should set `train_model` to `false` in `input.yaml` to run the code in parallel.
MPI parallelization is supported for only `generate_feature` and `preprocess`.

Please let me know if you have any problems during running.
Also, I recommend you to visit our new [documentation page](https://simple-nn-v2.readthedocs.io/en/latest).

Best regards,
Jisu.

**Comment by vsumaria** [comment]
Yes, I went through the documentation and realized that parallelization was meant for only generate_feature and preprocess. 

I have one issue in the evaluation process (using the same SiO2 example). I am now sure what I am doing wrong, but I changed the training input.yaml nodes to 15-15. But while evaluating using the "continue: checkpoint_bestmodel.pth.tar", I get the following error:

size mismatch for nets.Si.lin.lin_0.weight: copying a param with shape torch.Size([15, 70]) from checkpoint, the shape in current model is torch.Size([30, 70]).

**Comment by vsumaria** [comment]
Ok, I realized my mistake, I just had to add the nodes and other NN settings back in the evaluation input.yaml as well. 

Thanks!

**Comment by vsumaria** [comment]
Few more issues. I was not trying to use for a more complex system I have been wanting to use the code for. It has 5 elements, as a result the number of SF are high. When trying to preprocess the data, it is leading to segmentation issues (memory problem). 

Any suggestions on how to work with such large systems?

**Comment by vsumaria** [comment]
It would be good if there would be a way to estimate the memory usage for the preprocessing, that can help identify what parallel settings should be used.

**Comment by JisuJung928** [comment]
The memory of preprocessed data increases linearly and squarely as the number of SFs and atoms, respectively.
Therefore, if you suffer from a memory problem, please reduce the number of atoms in the system first.
Additionally, the less number of SFs sometimes provides enough accuracy in the application (radial < 8 and angular < 16).
Please test whether your system really needs all SFs or not.
(We solved the LAMMPS memory problem that occurred at more than 5 elements before.)

**Comment by JisuJung928** [comment]
I agree with your suggestion. We will consider how to estimate memory usage before preprocessing.

Thanks!
